### PR TITLE
feat(reddit): `hide-subreddit-banner` patch & bump patches compatibility to `v2023.05.0`

### PR DESCRIPTION
**Pull request by kazimmt**
- Bump patches compatibility to v2023.05.0
- Hides banner ads from the comment of subreddits.

### CODE REVIEW

**Comment by kazimmt** [comment]
Credits goes to: @inotia00 

Example of Layout Ads:

File: src/main/kotlin/app/revanced/patches/reddit/ad/general/annotations/GeneralAdsCompatibility.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import app.revanced.patcher.annotation.Package
 
 @Compatibility(
     [Package(
-        "com.reddit.frontpage", arrayOf("2022.43.0")
+        "com.reddit.frontpage", arrayOf("2023.05.0")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/reddit/ad/general/annotations/GeneralAdsCompatibility.kt:8]: Why has the version be replaced instead of added?

Comment (kazimmt) [reply src/main/kotlin/app/revanced/patches/reddit/ad/general/annotations/GeneralAdsCompatibility.kt:8]: > 

To match with Layout Ads
Should I keep old version too?

File: src/main/kotlin/app/revanced/patches/reddit/ad/layout/patch/LayoutAdsPatch.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+@Name("layout-reddit-ads")
+@Description("Removes layout ads from the Reddit frontpage and subreddits.")
+@LayoutAdsCompatibility
+@Version("0.0.1")
+class LayoutAdsPatch : ResourcePatch {
+    override fun execute(context: ResourceContext): PatchResult {
+        arrayOf(
+            "height",
+            "width"
+        ).forEach { replacements ->
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/reddit/ad/layout/patch/LayoutAdsPatch.kt:23]:
```suggestion
        ).forEach { attribute ->
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by kazimmt** [comment]
@oSumAtrIX all fix now

**Comment by oSumAtrIX** [comment]
Thanks.

**Pull request merged**